### ISSUE TITLE
MLT master fails to build with ffmpeg 7.0

### ISSUE THREAD

**Issue by berolinux** [mltframework/mlt#980]
FFmpeg 7.0 has dropped the old channel_layout API.
MLT currently (even on master) uses this API unconditionally.

**Comment by berolinux** [comment]
For reference, this https://github.com/owntone/owntone-server/pull/1536 is how another project using the same now-removed APIs has fixed it

**Comment by ddennedy** [comment]
@bmatherly I see you assigned this to yourself, but I started working on it hours ago without yet assigning. I have something compiling now that needs further work. I will push it into a branch.

**Comment by bmatherly** [comment]
@ddennedy sounds good. I'll stay out of your way. I'm happy to help testing.

**Comment by ddennedy** [comment]
Branch is https://github.com/mltframework/mlt/tree/ffmpeg7 , which builds now with some caveats for some other things outside of ch_layout API but also it crashes in Shotcut inside `link_swresample.c`. `melt -chain ...` reports
```
[SWR @ 0x78e154e0b680] Input channel layout "" is invalid or unsupported.
[link swresample] Cannot initialize context
```

**Comment by bmatherly** [comment]
Great! Do you want me to "take it from here"? Or just look into this specific crash?

**Comment by ddennedy** [comment]
Yes, I can use some help with this. I have done all I will do with it for at least a few days.
